scripts/hudson/quickstart.sh

```diff
@@ -156,7 +156,49 @@ function build_apache-karaf {
     exit -1
   fi
 }
+function clone_as {
+  REPO="https://github.com/wildfly/wildfly.git"
+  echo "Cloning AS sources from $REPO"
 
+  cd $WORKSPACE
+
+  if [ -d jboss-as ]; then
+    rm -rf jboss-as # start afresh
+  fi
+
+  git clone $REPO jboss-as || fatal "git clone $REPO failed"
+
+  if [ $REDUCE_SPACE = 1 ]; then
+    echo "Deleting git dir to reduce disk usage"
+    rm -rf .git
+  fi
+
+  cd $WORKSPACE
+}
+function build_as {
+  AS_BRANCH=27.0.0.Alpha1
+  echo "Building WildFly $AS_BRANCH"
+
+  cd $WORKSPACE/jboss-as
+
+  git checkout $AS_BRANCH || fatal "git checkout $AS_BRANCH failed"
+
+  # building WildFly
+  [ "$_jdk" -lt 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m -XX:+UseConcMarkSweepGC $MAVEN_OPTS"
+  [ "$_jdk" -ge 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m $MAVEN_OPTS"
+  JAVA_OPTS="-Xms1303m -Xmx1303m -XX:MaxMetaspaceSize=512m $JAVA_OPTS" ./build.sh clean install -B -DskipTests -Dts.smoke=false $IPV6_OPTS -Dversion.org.jboss.narayana=${NARAYANA_CURRENT_VERSION} "$@"
+  [ $? -eq 0 ] || fatal "AS build failed"
+
+  WILDFLY_VERSION_FROM_JBOSS_AS=`awk '/wildfly-parent/ { while(!/<version>/) {getline;} print; }' ${WORKSPACE}/jboss-as/pom.xml | cut -d \< -f 2|cut -d \> -f 2`
+  echo "AS version is ${WILDFLY_VERSION_FROM_JBOSS_AS}"
+  JBOSS_HOME=${WORKSPACE}/jboss-as/build/target/wildfly-${WILDFLY_VERSION_FROM_JBOSS_AS}
+  export JBOSS_HOME=`echo  $JBOSS_HOME`
+
+  # init files under JBOSS_HOME before AS TESTS is started
+  init_jboss_home
```

Comment:
Where is this function defined?

Reply:
Thanks @mmusgrov . I hope my last commit fixes that.